platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +182,26 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try {
+                TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream();
+                IOUtils.copy(is, tfbos);
+                is = tfbos.asByteSource().openStream();
+
+                Parser parser = new AutoDetectParser();
+                ToXMLContentHandler xmlContentHandler = new ToXMLContentHandler();
+                Metadata metadata = new Metadata();
+                parser.parse(is, xmlContentHandler, metadata, new ParseContext());
+                is.close();
```

Comment:
If something errors, this stream is going to be left unclosed. Need to probably open this in a try-with-resources.